runtime/src/core/memory.js

```diff
@@ -1,26 +1,817 @@
-import { mallocJSModule } from './malloc';
+import { mallocJSModule, printAllocations } from './malloc';
+import { getTagType, tagToString, heapTagToString, GRAIN_CONST_TAG_TYPE, GRAIN_NUMBER_TAG_TYPE, GRAIN_TUPLE_TAG_TYPE, GRAIN_GENERIC_HEAP_TAG_TYPE, GRAIN_LAMBDA_TAG_TYPE,
+         GRAIN_ADT_HEAP_TAG } from './tags';
+import { toHex, toBinary } from '../utils/utils';
+import treeify from 'treeify';
 
+export const TRACE_MEMORY = false;
+let TRACE_OVERRIDE = false;
+
+// Graph coloring
+const GREEN = 'GREEN';
+const RED = 'RED';
+const BLUE = 'BLUE';
+const BLACK = 'BLACK';
```

Comment:
Might be better to use Symbols for this.

Reply:
Fixed.